src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DefaultRazorComponentSearchEngine.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.Language.Intermediate;
+using Microsoft.AspNetCore.Razor.Language.Syntax;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Razor;
+using Microsoft.CodeAnalysis.Razor.ProjectSystem;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer
+{
+    internal class DefaultRazorComponentSearchEngine : RazorComponentSearchEngine
+    {
+        private readonly ProjectSnapshotManager _projectSnapshotManager;
+
+        public DefaultRazorComponentSearchEngine(ProjectSnapshotManager projectSnapshotManager)
+        {
+            _projectSnapshotManager = projectSnapshotManager ?? throw new ArgumentNullException(nameof(projectSnapshotManager));
+        }
+
+        /// <summary>Search for a component in a project based on its tag name and fully qualified name.</summary>
+        /// <remarks>
+        /// This method makes several assumptions about the nature of components. First, it assumes that a component
+        /// a given name `Name` will be located in a file `Name.razor`. Second, it assumes that the namespace the
+        /// component is present in has the same name as the assembly its corresponding tag helper is loaded from.
+        /// Implicitly, this method inherits any assumptions made by TrySplitNamespaceAndType.
+        /// </remarks>
+        /// <param name="tagHelper">A TagHelperDescriptor to find the corresponding Razor component for.</param>
+        /// <param name="documentSnapshot">A document snapshot if found, null otherwise.</param>
+        /// <returns>A tuple containing the document URI and related resources loaded during search.</returns>
+        public override bool TryLocateComponent(TagHelperDescriptor tagHelper, out DocumentSnapshot documentSnapshot)
+        {
+            DefaultRazorTagHelperBinderPhase.ComponentDirectiveVisitor.TrySplitNamespaceAndType(tagHelper.Name, out _, out var typeSpan);
+            var typeName = tagHelper.Name.Substring(typeSpan.Start, typeSpan.Length);
+
+            foreach (var project in _projectSnapshotManager.Projects)
+            {
+                if (!project.FilePath.EndsWith($"{tagHelper.AssemblyName}.csproj", FilePathComparison.Instance))
```

Comment:
Need to mention in the comments that we are assuming that the project file name is the same as the assembly name. 

@NTaylorMullen, given that it is not uncommon to have different assembly name and csproj file name, are we okay making this assumption?

Reply:
We can eventually change it but for now we're ok with it. It just means you can't rename something in a project with a different assembly name